src/analyseReport.js

```diff
@@ -1,71 +1,42 @@
 const fs = require("fs");
-
 let specs = [];
 const browsers = ["chrome", "firefox"];
 const defaultBrowser = "chrome"; // I will use it in case of there is no browser in the title.
 
 function getBrowserFromTitle(title) {
-  title = title.toLowerCase();
-  for (let browser of browsers) {
-    if (title.includes(browser)) {
-      return browser;
-    }
-  }
+  return browsers.find(browser => title.toLowerCase().includes(browser))
 }
 
 function orderBasedOnBrowserDuration(browser) {
-  let aDuration, bDuration;
   return specs.sort(function (a, b) {
-    for (let i = 0; i < a.data.length; i++) {
-      if (a.data[i].browser === browser) {
-        aDuration = a.data[i].duration;
-        break;
-      }
-    }
-
-    for (let i = 0; i < b.data.length; i++) {
-      if (b.data[i].browser === browser) {
-        bDuration = b.data[i].duration;
-        break;
-      }
-    }
-
+    let aDuration = a.data.find( item => item.browser === browser ).duration
+    let bDuration = b.data.find( item => item.browser === browser ).duration
     return aDuration - bDuration;
   });
 }
 
 function intiateSpecData(specName) {
-  for (let spec of specs) {
-    if (spec["specName"] === specName) {
-      return;
-    }
+  if (!specs.find( spec => spec.specName === specName )) {
```

Comment:
@0xIslamTaha for creating arrays or adding to array you can spread `...` array and use lodash uniq to make it unique. Or you can try Array.map or Array.forEach 🤔

Reply:
@RadekM86
I need to intialize it only for the first time so that is why I'm using find and map

Reply:
Ok 👍